src/Birkhoff_polytope.jl

```diff
@@ -0,0 +1,453 @@
+    row_min_sum = zeros(n)  # minimum possible sum for each row
+    row_max_sum = zeros(n)  # maximum possible sum for each row
+    col_min_sum = zeros(n)  # minimum possible sum for each column
+    col_max_sum = zeros(n)  # maximum possible sum for each column
+
+    # Process each integer variable
+    for idx in eachindex(blmo.int_vars)
+        var_idx = blmo.int_vars[idx]
+
+        # Convert linear index to (row, col) based on storage format
+        if blmo.append_by_column
+            j = ceil(Int, var_idx / n0)  # column index
+            i = Int(var_idx - n0 * (j - 1))  # row index
+        else
+            i = ceil(Int, var_idx / n0)  # row index  
+            j = Int(var_idx - n0 * (i - 1))  # column index
+        end
+
+        # Add bounds to row and column sums
+        row_min_sum[i] += blmo.lower_bounds[idx]
+        row_max_sum[i] += blmo.upper_bounds[idx]
+        col_min_sum[j] += blmo.lower_bounds[idx]
+        col_max_sum[j] += blmo.upper_bounds[idx]
+    end
+
+    # Check feasibility: each row and column must be able to sum to exactly 1
+    for i = 1:n0
+        # Check row sum constraints
+        if row_min_sum[i] > 1 + eps() || row_max_sum[i] < 1 - eps()
+            return Boscia.INFEASIBLE
+        end
+
+        # Check column sum constraints  
+        if col_min_sum[i] > 1 + eps() || col_max_sum[i] < 1 - eps()
+            return Boscia.INFEASIBLE
+        end
+    end
+
+    return Boscia.OPTIMAL
+end
+
+function Boscia.update_model(blmo::BirkhoffBLMO)
+    dim = blmo.dim
+    n = dim^2
+
+    # fixed_to_one_vars = [v for v in blmo.int_vars if blmo.lower_bounds[v] == 1.0]
+
+    fixed_to_one_vars = blmo.int_vars[blmo.lower_bounds.==1.0]
+    empty!(blmo.fixed_to_one_rows)
+    empty!(blmo.fixed_to_one_cols)
+    for fixed_to_one_var in fixed_to_one_vars
+        q = (fixed_to_one_var - 1) ÷ dim
+        r = (fixed_to_one_var - 1) - q * dim
+        if blmo.append_by_column
+            i = r + 1
+            j = q + 1
+        else
+            i = q + 1
+            j = r + 1
+        end
+        push!(blmo.fixed_to_one_rows, i)
+        push!(blmo.fixed_to_one_cols, j)
+    end
+
+    nfixed = length(blmo.fixed_to_one_rows)
+    nreduced = dim - nfixed
+
+    # stores the indices of the original matrix that are still in the reduced matrix
+    index_map_rows = fill(1, nreduced)
+    index_map_cols = fill(1, nreduced)
+    idx_in_map_row = 1
+    idx_in_map_col = 1
+    for orig_idx = 1:dim
+        if orig_idx ∉ blmo.fixed_to_one_rows
+            index_map_rows[idx_in_map_row] = orig_idx
+            idx_in_map_row += 1
+        end
+        if orig_idx ∉ blmo.fixed_to_one_cols
+            index_map_cols[idx_in_map_col] = orig_idx
+            idx_in_map_col += 1
+        end
+    end
+
+    empty!(blmo.index_map_rows)
+    empty!(blmo.index_map_cols)
+    append!(blmo.index_map_rows, index_map_rows)
+    append!(blmo.index_map_cols, index_map_cols)
+end
```

Comment:
I am not convinced that we need this function. I think you can do this in the add, delete and set bound functions.